### PR TITLE
chore(action): use cd api for dev and prod based on environment

### PR DESCRIPTION
**Pull request by guilbep**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,46 +2,108 @@
...
+          ENAC_IT4R_CD_deployment_secret: ${{ secrets.DEPLOYMENT_SECRET }}
+          ENAC_IT4R_CD_deployment_id: ${{ secrets.DEPLOYMENT_ID }}
+```
+
+## Polling for status with a timeout
+
+This script makes a POST request to a given URL until the status is `finished` or `error`
+Initially created for checking if a enac deployment is complete (via enacit-ansible)
+
+### Inputs
```
Comment (ymarcon) [inline README.md:63]: The following sections should be at the next sub-level ####

Comment (guilbep) [reply README.md]: Thanks! I moved everything!

File: README.md [hunk 2]
```diff
@@ -2,46 +2,108 @@
...
-# https://github.com/EPFL-ENAC/github-actions-runner#readme
+# https://github.com/EPFL-ENAC/epfl-enac-deploy-action#readme
 name: deploy-test
 
-on:
+"on":
   push:
     branches:
-      - develop
+      - dev
```
Comment (sbancal) [inline README.md:50]: I think ENAC-IT's default is `develop`.

File: README.md [hunk 3]
```diff
@@ -2,46 +2,108 @@
...
+
+### `ENAC_IT4R_CD_deployment_secret`
+  - The secret associated with the deployment_id
+  - only viable for that unique id
+
+### `ENAC_IT4R_CD_deployment_id`
+
+The unique id associated with the deployment:
+  - maybe the fqdn where it will be deployed
+  - maybe a unique id (uuid4 or anything)
```
Comment (sbancal) [inline README.md:77]: It's value will be chosen and given by ENAC-IT's sysadmins at the same time as the `ENAC_IT4R_CD_deployment_secret`.
Yes, it'll be the fqdn, but I'd simply say:

The unique deployment id received by ENAC-IT's sysadmins.

File: README.md [hunk 4]
```diff
@@ -2,46 +2,108 @@
...
+          ENAC_IT4R_CD_deployment_id: ${{ secrets.DEPLOYMENT_ID }}
+```
+
+## Inputs
+
+### `ENAC_IT4R_CD_environment`
+
+The environement to deploy to: dev or prod
+
+### `ENAC_IT4R_CD_deployment_secret`
```
Comment (sbancal) [inline README.md:69]: As said in an other comment, I'd put this below `ENAC_IT4R_CD_deployment_id`

File: README.md [hunk 5]
```diff
@@ -2,46 +2,108 @@
...
+
+```
+{
+  status: "finished" | "error",
+  output: "aeae",
+}
+```
+
+Script exit with 1 when status is `error` and 0 when `finished`
+Exit with 1 if timeout reached before receiving `error` or `finished` as status
```
Comment (sbancal) [inline README.md:109]: What about something distinct ?

File: action.yml [hunk 6]
```diff
@@ -1,17 +1,91 @@
...
+---
+name: "EPFL ENAC-IT4R Deploy"
+description: "EPFL ENAC-IT4R Deployment"
 inputs:
-  token:
-    description: "GitHub Token"
+  ENAC_IT4R_CD_environment:
+    description: "ENAC IT Continuous api environment"
     required: true
+  ENAC_IT4R_CD_deployment_secret:
```
Comment (sbancal) [inline action.yml:8]: would put below `ENAC_IT4R_CD_deployment_id`

File: main.sh [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+      echo "./main.sh [options]"
+      echo " "
+      echo "options:"
+      echo "-h, --help           Show brief help"
+      echo "--url=URL            url to poll"
+      echo "--APP_NAME=APP_NAME     repository name"
+      echo "--API_CD_TOKEN=API_CD_TOKEN         token given by cd"
+      echo "--JOB_ID=JOB_ID          job id given by cd api"
+      echo "--interval=INTERVAL  Interval between each call, in seconds"
+      echo "--timeout=TIMEOUT    Timeout before stop polling, in seconds"
```
Comment (sbancal) [inline main.sh:17]: timeout is not implemented in this script.

File: main.sh [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+done
+
+function poll_status {
+  echo "$JOB_ID"
+  while true;
+  do
+    curl -X POST -H "Content-Type: application/json" -d "{\"key\":\"$API_CD_TOKEN\",\"name\":\"$APP_NAME\",\"job_id\":\"$JOB_ID\"}" $url -s > temp.json;
+    export iatus=$(cat temp.json | jq -cr '.status');
+    echo "$(date +%H:%M:%S): status is $iatus";
+    cat temp.json | jq -cr '.output';
```
Comment (sbancal) [inline main.sh:53]: This is the line that will produce the output. Right ?

**Review by sbancal — DISMISSED** [review]
See my comment.

Question: Why not put all the logic in one place? Here some of the story is in `action.yml` and the last steps are in `main.sh`.

File: README.md [hunk 9]
```diff
@@ -2,46 +2,108 @@
 
 EPFL ENAC-IT Continuous Deployment
 
+## Create two secrets in your repository
+
+- under your repository settings in /settings/secrets/actions
+- Add DEPLOYMENT_SECRET and DEPLOYMENT_ID given by the linux enacit-linux-sysadmins@epf.ch
```
[the remark this answers is not in the document]
Comment (guilbep) [reply README.md]: Merci

File: main.sh (hunk 8 above)
Comment (guilbep) [reply main.sh:52]: Thank you for the feedback :)

**Pull request merged**